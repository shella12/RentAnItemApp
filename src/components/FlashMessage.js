import React, { useState } from 'react';
import PropTypes from 'prop-types';

const FlashMessage = ({ message, duration }) => {
  const [visible, setVisible] = useState(true);

  setTimeout(() => {
    setVisible(false);
  }, duration);

  return visible ? <div className="flash-message">{message}</div> : null;
};

FlashMessage.propTypes = {
  message: PropTypes.string.isRequired,
  duration: PropTypes.number.isRequired,
};
export default FlashMessage;
