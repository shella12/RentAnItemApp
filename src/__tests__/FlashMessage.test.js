import renderer, { act } from 'react-test-renderer';
import '@testing-library/jest-dom';
import React from 'react';
import { waitFor } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import FlashMessage from '../components/FlashMessage';

describe('test render', () => {
  test('Flash House should match snapshoot', async () => {
    let tree;
    act(() => {
      tree = renderer.create(
        <React.StrictMode>
          <MemoryRouter MemoryRouter initialEntries={['/']}>
            <FlashMessage message="Test Navbar" duration={20} />
          </MemoryRouter>
        </React.StrictMode>,
      );
    });

    await waitFor(() => {
      expect(tree.toJSON()).toMatchSnapshot();
    });
  });
});
