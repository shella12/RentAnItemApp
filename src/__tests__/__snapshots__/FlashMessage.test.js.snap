// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test render Flash House should match snapshoot 1`] = `
<div
  className="flash-message"
>
  Test Navbar
</div>
`;
